### PR TITLE
checkin/models.py: Add docstrings

### PR DESCRIPTION
**Pull request by mahoyen**
Opening a draft pull request for easier follow up of my work. 

First time writing docstrings and I starting doing it in `checkin/models.py`. I have currently only added for one function, but will continue to write docstrings for the whole file before it is ready for review.

Please give feedback if you have any!

### CODE REVIEW

**Comment by mahoyen** [comment]
Now I have added docstrings to all the classes in checkin/models.py.

I am unsure of `:var voters: Profile model of who has voted for suggested skill.` in the class `SuggestSkill` and if `:func ` is a thing I can use to describe the functions in a class.

Other than that I think it is ready for review.

File: checkin/models.py [hunk 1]
```diff
@@ -3,6 +3,15 @@
 
 
 class Skill(models.Model):
+    """Skill with image
```
Comment (zootos) [inline checkin/models.py]: Unless the docstring is a single linem start writing on the line after `"""`, that is:
```python
"""This is a description"""
```
and
```python
"""
This is a description

:var abc: Var description
"""
```
are fine, while
```python
"""This is a description

:var abc: Var description
"""
```
is not fine.

File: checkin/models.py [hunk 2]
```diff
@@ -3,6 +3,15 @@
...
+
+    :var title: Skill name in Norwegian
+    :var title_en: Skill name in English
+    :var image: Skill image
+
+    :func __str__: returns skill name
```
Comment (zootos) [inline checkin/models.py]: We don't really use the `:func abc:`

File: checkin/models.py [hunk 3]
```diff
@@ -11,12 +20,30 @@ def __str__(self):
         return self.title
 
     def locale_title(self, language_code):
+        """Returns skill name in correct language.
+
+        Only 'nb' is recognised, everything else will return the english skill name.
+
+        :param language_code: Two letter string with language code
+        :return: skill name in correct language
+
```
Comment (zootos) [inline checkin/models.py]: This empty line should not be here

File: checkin/models.py [hunk 4]
```diff
@@ -11,12 +20,30 @@ def __str__(self):
...
+        """
+
```
Comment (zootos) [inline checkin/models.py]: Same goes for this empty line

File: checkin/models.py [hunk 5]
```diff
@@ -11,12 +20,30 @@ def __str__(self):
         return self.title
 
     def locale_title(self, language_code):
+        """Returns skill name in correct language.
+
+        Only 'nb' is recognised, everything else will return the english skill name.
+
+        :param language_code: Two letter string with language code
```
Comment (zootos) [inline checkin/models.py]: This should rather be "A string representing an ISO 639-1 language code"

File: checkin/models.py [hunk 6]
```diff
@@ -11,12 +20,30 @@ def __str__(self):
         return self.title
 
     def locale_title(self, language_code):
+        """Returns skill name in correct language.
+
+        Only 'nb' is recognised, everything else will return the english skill name.
```
Comment (zootos) [inline checkin/models.py]: Should rather be "Supports Norwegian Bokmål and English, defaulting to English for all other language codes."

File: checkin/models.py [hunk 7]
```diff
@@ -11,12 +20,30 @@ def __str__(self):
...
+
+        :param language_code: Two letter string with language code
+        :return: skill name in correct language
```
Comment (zootos) [inline checkin/models.py]: "skill name in correct language" should probably be "skill name in the given language" and capitalized.

File: checkin/models.py [hunk 8]
```diff
@@ -11,12 +20,30 @@ def __str__(self):
         return self.title
 
     def locale_title(self, language_code):
+        """Returns skill name in correct language.
```
Comment (zootos) [inline checkin/models.py]: Personally I am not a big fan of using returns here, as that generally means a repeat of what is said at `:return:`. Should rather be something like "Finds the title of the skill in the given language". Should also be combined with the line below.

File: checkin/models.py [hunk 9]
```diff
@@ -11,12 +20,30 @@ def __str__(self):
...
+
         if language_code == "nb":
             return self.title
         return self.title_en
 
 
 class Profile(models.Model):
+    """Profile model
+
+    :var user: User model
```
Comment (zootos) [inline checkin/models.py]: "The user to which this profile belongs"

File: checkin/models.py [hunk 10]
```diff
@@ -11,12 +20,30 @@ def __str__(self):
...
+    :var image: Profile picture
+    :var on_make: boolean, if at Makerverkstedet
+    :var last_checkin: DateTimeField, last time at Makerverkstedet
```
Comment (zootos) [inline checkin/models.py]: No need to mention the type of field, this can easily be seen below.

File: checkin/models.py [hunk 11]
```diff
@@ -29,6 +56,14 @@ def __str__(self):
 
 
 class UserSkill(models.Model):
+    """Model for user's skill
+
+    :var profile: Profile model
+    :var skill: skill model
```
Comment (zootos) [inline checkin/models.py]: This docstring does not provide any new information. At least describe the purpose of the model

File: checkin/models.py [hunk 12]
```diff
@@ -58,6 +104,14 @@ def __str__(self):
         return self.title
 
     def locale_title(self, language_code):
+        """Returns skill name in correct language.
+
+        Only 'nb' is recognised, everything else will return the english skill name.
```
Comment (zootos) [inline checkin/models.py]: Same as for skill above

File: checkin/models.py [hunk 13]
```diff
@@ -70,6 +124,15 @@ class Meta:
 
 
 class RegisterProfile(models.Model):
+    """Model for registrating profile
+
+        :var card_id: Charfield with student card number.
+        :var last_scan: DateTimeField with time of last scan
```
Comment (zootos) [inline checkin/models.py]: Why this indenting?

**Comment by mahoyen** [comment]
Added the corrections mentioned in the comments. I am not sure of the one-line summary of the models as I don't know how they are used.

File: checkin/models.py [hunk 14]
```diff
@@ -11,12 +18,28 @@ def __str__(self):
         return self.title
 
     def locale_title(self, language_code):
+        """
+        Finds the title of the skill in the given language
+
```
Comment (zootos) [inline checkin/models.py]: Combine these two sentences, both describing the method, into one line. If the line becomes longer than 120 characters (counting each tab as 4 characters), place the remaining content on the next line.

File: checkin/models.py [hunk 15]
```diff
@@ -3,6 +3,13 @@
 
 
 class Skill(models.Model):
+    """
+    Skill with image
+
+    :var title: Skill name in Norwegian
+    :var title_en: Skill name in English
```
Comment (zootos) [inline checkin/models.py]: Remove the vars here, they don't add anything. That is, the comment makes as much sense by just having it as 
```python
"""Defines a skill that a user can have."""
```

File: checkin/models.py [hunk 16]
```diff
@@ -11,12 +18,28 @@ def __str__(self):
...
+        :return: Skill name in given language
+        """
         if language_code == "nb":
             return self.title
         return self.title_en
 
 
 class Profile(models.Model):
+    """
+    Profile model, storing status of last been at Makerverkstedet.
```
Comment (zootos) [inline checkin/models.py]: "A model for user profiles, storing the status of the their last visit to Makerverkstedet"

File: checkin/models.py [hunk 17]
```diff
@@ -11,12 +18,28 @@ def __str__(self):
...
+
+    :var user: The user to which this profile belongs
+    :var image: Profile picture
+    :var on_make: If at Makerverkstedet
```
Comment (zootos) [inline checkin/models.py]: "Indicates if the user is at Makerverkstedet"

File: checkin/models.py [hunk 18]
```diff
@@ -29,6 +52,13 @@ def __str__(self):
 
 
 class UserSkill(models.Model):
+    """
+    Model for user's skill, storing skill and skill level of skill
+
+    :var profile: Profile of the user which this skill belongs to
+    :var skill: skill model
+    :var skill_level: Three levels, 1=beginner, 3=expert
```
Comment (zootos) [inline checkin/models.py]: The user's skill level, can be one of three levels: "beginner", "intermediate" or "expert"

File: checkin/models.py [hunk 19]
```diff
@@ -29,6 +52,13 @@ def __str__(self):
 
 
 class UserSkill(models.Model):
+    """
+    Model for user's skill, storing skill and skill level of skill
+
+    :var profile: Profile of the user which this skill belongs to
+    :var skill: skill model
```
Comment (zootos) [inline checkin/models.py]: "The skill which the object indicates the user's skill"

File: checkin/models.py [hunk 20]
```diff
@@ -29,6 +52,13 @@ def __str__(self):
 
 
 class UserSkill(models.Model):
+    """
+    Model for user's skill, storing skill and skill level of skill
+
+    :var profile: Profile of the user which this skill belongs to
```
Comment (zootos) [inline checkin/models.py]: "The profile of the user"

File: checkin/models.py [hunk 21]
```diff
@@ -29,6 +52,13 @@ def __str__(self):
 
 
 class UserSkill(models.Model):
+    """
+    Model for user's skill, storing skill and skill level of skill
```
Comment (zootos) [inline checkin/models.py]: "A model for indicating a user's skill level"

File: checkin/models.py [hunk 22]
```diff
@@ -58,6 +97,14 @@ def __str__(self):
         return self.title
 
     def locale_title(self, language_code):
+        """
+        Finds the title of the skill in the given language
+
+        Supports Norwegian Bokmål and English, defaulting to English for all other language codes.
```
Comment (zootos) [inline checkin/models.py]: Combine the sentences as in the other comment

**Comment by kalkins** [comment]
@mahoyen Are you going to finish this PR? If not, I'll implement the changes from the comments and merge it.

**Comment by mahoyen** [comment]
Made the changes now.

**Pull request merged**